### PR TITLE
vertexai[patch]: remove allOf from function schema

### PR DESCRIPTION
**Pull request by baskaryan**
patch for #95

### CODE REVIEW

**Comment by baskaryan** [comment]
@alx13 @lkuligin

File: libs/vertexai/langchain_google_vertexai/functions_utils.py [hunk 1]
```diff
@@ -94,6 +94,31 @@ class Config:
...
 
 
+def _replace_all_ofs(schema: Dict[str, Any]) -> Dict[str, Any]:
+    """Assumes dereferenced schema."""
+    new_schema: Dict[str, Any] = {}
+    for k, v in schema.items():
+        if isinstance(v, dict) and "allOf" in v:
+            if isinstance(v["allOf"], list) and len(v["allOf"]) == 1:
+                obj = dict(v["allOf"][0])
+                obj["title"] = v.get("title", obj.get("title", ""))
```
Comment (alx13) [inline libs/vertexai/langchain_google_vertexai/functions_utils.py:104]: `title` field is now not supported - we are dropping it

File: libs/vertexai/langchain_google_vertexai/functions_utils.py [hunk 2]
```diff
@@ -94,6 +94,31 @@ class Config:
...
+                obj["description"] = " ".join(
+                    (v.get("description", ""), obj.get("description", ""))
```
Comment (alx13) [inline libs/vertexai/langchain_google_vertexai/functions_utils.py:106]: I'm not sure about joining descriptions.

This code:
```python
class Node(BaseModel):
    """This is a node"""
    id: str
    type: str

class Relationship(BaseModel):
    source: Node
    target: Node = Field(..., description="This is a target node")
```
will produce following definition for target property:
```json
{
  "target": {
    "title": "Target",
    "description": "This is a target node This is a node",
    "type": "object",
    "properties": {
      "id": {
        "title": "Id",
        "type": "string"
      },
      "type": {
        "title": "Type",
        "type": "string"
      }
    },
    "required": ["id", "type"]
  }
}
```

Comment (baskaryan) [reply libs/vertexai/langchain_google_vertexai/functions_utils.py:106]: yea not ideal but is there a better generic solution? both descriptions could have important (and distinct) info

Comment (alx13) [reply libs/vertexai/langchain_google_vertexai/functions_utils.py:106]: I would consider coalescing i. e. field description should take precedence

```
class Relationship(BaseModel):
    source: Node
    target: Node = Field(..., description="This is a target node")
```

over model description:
```
class Node(BaseModel):
    """This is a node"""
    id: str
    type: str
```

And could you please describe reasoning and behaviour in the function doc string?

Comment (alx13) [reply libs/vertexai/langchain_google_vertexai/functions_utils.py:106]: @lkuligin what is your take on that?

**Pull request closed without merging**